eth/downloader/downloader.go

```diff
@@ -325,52 +367,50 @@ func (d *Downloader) UnregisterPeer(id string) error {
 	// Unregister the peer from the active peer set and revoke any fetch tasks
 	glog.V(logger.Detail).Infoln("Unregistering peer", id)
 	err = d.peers.Unregister(id)
+
```

Comment:
Here's a tiny difference from EF AFAIR -- they only abort sync from master _after_ the unregister `err` is checked. But it seems to me that whether or not the unregister failed (which I've only ever seen with error `peer not registered`), that the sync should be aborted. Even _especially_ if the peer isn't registered for whatever reason.